### DISCUSSION TITLE
How to upload documents to Azure OpenAI in an assistant?

### DISCUSSION THREAD

**Discussion by roldengarm** [microsoft/semantic-kernel#8063] [Q&A] [answered]
Probably very simple question, but I'm a bit stuck. We're using SK for a chat bot including plugins. What I want to do now, is have the ability for the user to upload a document (always DOCX), and that the assistant performs actions on the document. So, I have to upload the whole document.
I've looked through the documentation and old PR's. I can find that an OpenAIFileService was added, but then later [deleted](https://github.com/microsoft/semantic-kernel/pull/7195).
The Azure OpenAI SDK should support [file upload](https://github.com/microsoft/semantic-kernel/issues/6443) but not sure how to invoke that from my C# code.
I have a reference to the `Kernel` object and currently invoke e.g. `InvokePromptStreamingAsync` .
How can I also upload documents?

**Comment by moonbox3** [comment, marked as answer]
Hi @roldengarm, as of today's dotnet 1.18.0 RC1 release, one is able to use OpenAI Assistant V2 agents. You can use the following as an example on how to leverage the assistant's vector store and file search:

https://github.com/microsoft/semantic-kernel/blob/feature-connectors-openai/dotnet/samples/GettingStartedWithAgents/Step11_AssistantTool_FileSearch.cs

**Reply by roldengarm** [reply]
Thanks, that is awesome news @moonbox3 ! I had seen that, great that it's now available in SK.
In our particular usecase we need the original document though, not the indexed data. We want to ask GPT to rewrite the document based on certain instructions, e.g. to change the tone or change a name. 
I can't just send the raw text as then images/markup will be gone.

Any suggestions please let me know 👍

**Reply by moonbox3** [reply]
@alliscode do you have thoughts about this use case?

**Comment by sophialagerkranspandey** [comment]
We'd recommend you extract the text and put placeholders for the images and other items to proceed. If you do need images/markup to improve the quality of the results we recommend using something like document intelligence in Azure. https://azure.microsoft.com/en-us/products/ai-services/ai-document-intelligence
